larq_compute_engine/core/bgemm_kernels_arm32.h

```diff
@@ -0,0 +1,527 @@
+//  |v1.s[0] ... v1.s[3] |  |v8.s[1]        ...         v11.s[1]  |
+//  |v2.s[0] ... v2.s[3] |  |v8.s[2]        ...         v11.s[2]  |
+//  |v3.s[0] ... v3.s[3] |  |v8.s[3]        ...         v11.s[3]  |
+//  \--------------------/  \--------------------------------------/
+//                                  int32 accumulators 4x4 block
+//
+// No attempt had been made so far at implementing the RUY_OPT_MAX_STREAMING
+// optimization for this kernel.
+
+// clang-format on
+
+void BinaryKernelNeonOutOfOrder32BP4x4(BinaryKernelParams<4, 4, std::uint32_t>& params) {
+  CheckOffsetsInKernelParams32BP(params);
+  ruy::profiler::ScopeLabel label(
+      "Binary Kernel (4x4) 32BP (kNeon, optimized for out-of-order cores)");
+
+  std::uint32_t* lhs_col_ptr = const_cast<std::uint32_t*>(params.lhs_base_ptr);
+  std::uint32_t* rhs_col_ptr = const_cast<std::uint32_t*>(params.rhs_base_ptr);
+  std::uint32_t* lhs_ptr = lhs_col_ptr;
+  std::uint32_t* rhs_ptr = rhs_col_ptr;
+
+  float* dst_col_ptr = params.dst_base_ptr;
+  float* dst_ptr = dst_col_ptr;
+  int row = params.start_row;
+  int col = params.start_col;
+
+  asm volatile(
+#define RUY_MAKE_ZERO(reg) "vmov.i32 " #reg ", #0\n"
+
+      "sub sp, sp, #" RUY_STR(RUY_STACK_OFFSET_SIZE) "\n"
+      //auto dst_col_ptr = params.dst_base_ptr;
+      "ldr r1, [%[params], #" RUY_STR(RUY_OFFSET_DST_BASE_PTR) "]\n"
+      "str r1, [sp, #" RUY_STR(RUY_STACK_OFFSET_DST_COL_PTR) "]\n"
+
+      //auto dst_ptr =  dst_col_ptr;
+      "ldr r2, [%[params], #" RUY_STR(RUY_OFFSET_DST_BASE_PTR) "]\n"
+      "str r2, [sp, #" RUY_STR(RUY_STACK_OFFSET_DST_PTR) "]\n"
+
+       //auto row = params.start_row
+      "ldr r1, [%[params], #" RUY_STR(RUY_OFFSET_START_ROW) "]\n"
+      "str r1, [sp, #" RUY_STR(RUY_STACK_OFFSET_ROW) "]\n"
+
+       //auto col = params.start_col
+      "ldr r3, [%[params], #" RUY_STR(RUY_OFFSET_START_COL) "]\n"
+      "str r3, [sp, #" RUY_STR(RUY_STACK_OFFSET_COL) "]\n"
+
+       //auto lhs_col_ptr = params.lhs_base_ptr
+      "ldr r1, [%[params], #" RUY_STR(RUY_OFFSET_LHS_BASE_PTR) "]\n"
+      "str r1, [sp, #" RUY_STR(RUY_STACK_OFFSET_LHS_COL_PTR) "]\n"
+
+       //auto rhs_col_ptr = params.rhs_base_ptr
+      "ldr r2, [%[params], #" RUY_STR(RUY_OFFSET_RHS_BASE_PTR) "]\n"
+      "str r2, [sp, #" RUY_STR(RUY_STACK_OFFSET_RHS_COL_PTR) "]\n"
+
+      // Clear accumulators.
+      RUY_MAKE_ZERO(q8)
+      RUY_MAKE_ZERO(q9)
+      RUY_MAKE_ZERO(q10)
+      RUY_MAKE_ZERO(q11)
+
+      // Load the first 64 bytes of LHS and RHS data.
+      "vld1.32 {d0, d1, d2, d3}, [%[lhs_ptr]]!\n"
+      "vld1.32 {d4, d5, d6, d7}, [%[lhs_ptr]]!\n"
+      "vld1.32 {d8, d9, d10, d11}, [%[rhs_ptr]]!\n"
+      "vld1.32 {d12, d13, d14, d15}, [%[rhs_ptr]]!\n"
+
+      // w1 is the number of levels of depth that we have already loaded
+      // LHS and RHS data for. The RHS is stored in col-wise. Therefore, for 32-bit elements,
+      // one register can hold 4 levels of depth.
+      "mov r1, #4\n"
+#if 1
```

Comment:
Is the `#if` needed?

Reply:
`#if` is not needed...I missed removing this debugging code